apps/site/lib/site_web/router.ex

```diff
@@ -264,6 +271,13 @@ defmodule SiteWeb.Router do
     get("/*path", CMSController, :page)
   end
 
+  defp basic_auth(conn, _) do
+    username = System.fetch_env!("BASIC_AUTH_USERNAME")
+    password = System.fetch_env!("BASIC_AUTH_PASSWORD")
+
+    Plug.BasicAuth.basic_auth(conn, username: username, password: password)
```

Comment:
suggestion: generally we put the environment variable values into the application config (`config/runtime.exs`) and then read it as necessary.
```elixir
# config/runtime.exs
config :site, SiteWeb.Router,
  cms_basic_auth: [
    username: System.fetch_env!("BASIC_AUTH_USERNAME"),
    password: System.fetch_env!("BASIC_AUTH_PASSWORD")
  ]

# router.ex
opts = Application.get_env(:site, SiteWeb.Endpoint)[:cms_basic_auth]
Plug.BasicAuth.basic_auth(conn, opts)
```

Reply:
yeah, I have to make another PR into devops to get these environment variables. i just wanted to make sure everything was actually going to work before finalizing on even using basic auth.